.github/workflows/release.yml

```diff
@@ -0,0 +1,67 @@
+name: Build, Test, Package and Push
+
+# Controls when the action will run.
+on:
+  release:
+    types: [created]
+
+  workflow_dispatch:
+    inputs:
+      release-url:
+        description: "The URL of the github release being created in OD"
+        required: true
+
+jobs:
+  build:
+    name: "Build and Package"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Set up JDK 8
+        uses: actions/setup-java@v2
+        with:
+          java-version: '8'
+          distribution: 'adopt'
+
+      - name: Build with gradle
+        run: ./gradlew build test
+
+      - name: Create Plugin Zip Deployment
+        id: create-package
+        run: |
+        ./gradlew distZip
+        PACKAGE_NAME=`./gradlew -q packageName`
+        echo "::set-output name=package-created::$PACKAGE_NAME"
+        version=`./gradlew properties | grep ^version | awk '{print $2;}'`
+        echo "::set-output name=version::$version"
+
+    - name: Push a package to Octopus Deploy 🐙
+      uses: OctopusDeploy/push-package-action@<version>
```

Comment:
Is "version" available here, I do not know how variables are scoped in GH actions

Reply:
yeah whoops - hard coded.